aurora/core.py

```diff
@@ -1103,18 +1145,93 @@ def calc_Zeff(self):
         Z = np.arange(Zmax + 1)
         self.delta_Zeff = nz * (Z * (Z - 1))[None, :, None]  # for each charge state
         self.delta_Zeff /= self.ne.T[:, None, :]
+        
+    def calc_taumain(self):
+        (
+            nz,
+            N_wall,
+            N_div,
+            N_pump,
+            N_out,
+            N_ret,
+            N_tsu,
+            N_dsu,
+            N_dsul,
+            rcld_rate,
+            rclp_rate,
+            rclw_rate,
+        ) = self.res
+        nz = nz.transpose(2, 1, 0)  # time,nZ,space
+        
+        # factor to account for cylindrical geometry:
+        circ = 2 * np.pi * self.Raxis_cm  # cm
+
+        # calculate total impurity density (summed over charge states)
+        total_impurity_density = np.nansum(nz, axis=1)  # time, space
+
+        # compute total number of particles in the plasma
+        N_main = grids_utils.vol_int(
+            total_impurity_density, self.rvol_grid, self.pro_grid, self.Raxis_cm,
+            rvol_max = None
+        )
+        
+        # compute fluxes leaving the plasma
+        edge_loss = N_tsu * circ
+        limiter_loss = N_dsul * circ
+        parallel_loss = N_dsu * circ
+        flux_out = edge_loss + limiter_loss + parallel_loss
+        
+        # calculate the impurity confinement time in the plasma
+        tau_main = np.divide(N_main, flux_out, out=np.zeros_like(N_main), where=flux_out!=0)
+        
+        return tau_main
+    
+    def calc_compression(self):
+        (
+            nz,
+            N_wall,
+            N_div,
+            N_pump,
+            N_out,
+            N_ret,
+            N_tsu,
+            N_dsu,
+            N_dsul,
+            rcld_rate,
+            rclp_rate,
+            rclw_rate,
+        ) = self.res
+        nz = nz.transpose(2, 1, 0)  # time,nZ,space
+        
+        # factor to account for cylindrical geometry:
+        circ = 2 * np.pi * self.Raxis_cm  # cm
+        
+        if not self.namelist["phys_volumes"]:    
+            raise ValueError("Calculating the divertor compression requires the physical volumes of the reservoirs to be defined.") 
+        
+        ones = np.ones_like(nz)
+        num = grids_utils.vol_int(nz[:,-1,:],self.rvol_grid,self.pro_grid,self.Raxis_cm,rvol_max=self.rvol_lcfs)
+        den = grids_utils.vol_int(ones[:,-1,:],self.rvol_grid,self.pro_grid,self.Raxis_cm,rvol_max=self.rvol_lcfs)
+        n_plasma = num/den
+        n_div = (N_div*circ)/self.vol_div
+        
+        compression = np.divide(n_div, n_plasma, out=np.zeros_like(n_div), where=n_plasma!=0)
+
+        return n_plasma, n_div, compression
+        
 
     def plot_resolutions(self):
         """Convenience function to show time and spatial resolution in Aurora simulation setup.
         """
         # display radial resolution
-        _ = grids_utils.create_radial_grid(self.namelist, plot=True)
+        _ = grids_utils.create_radial_grid(self.namelist, plot=False)
```

Comment:
@antonellozito what's the use of this `plot_resolutions` method now that you've set `plot=False` and the functions are not returning anything?